### PR TITLE
Add etcd caching for ListServices API

### PR DESCRIPTION
**Pull request by TheAlgo**
* Add Utilities for Etcd Client of Python
* Implement caching using etcd for `ListAllServices` API

### CODE REVIEW

File: api/routers/odin.py [hunk 1]
```diff
@@ -83,14 +87,17 @@ async def rollback_service(rollback_request: RollbackRequest):
 
 
 @router.get("/odin/services/", tags=["odin"])
-async def get_all_services():
+async def get_all_services(user_id: str):
     try:
-        service_list = Utils.getJson(Helm.listAllServices())
+        if Utils.checkIfKeyExist(etcd, '{user_id}:service_list'.format(user_id=user_id)):
```
Comment (redhood-97) [inline api/routers/odin.py:92]: 'user_id' is not used in user management. 
Use the acl-token instead.

Like 'acl:acl_token-{the_actual_acl_token}' : [list of services]

Aisa mapping maitain kar when inserting or search for an object.

Comment (redhood-97) [reply api/routers/odin.py:92]: @TheAlgo

File: api/util/utilityHelpers.py [hunk 2]
```diff
@@ -37,6 +37,24 @@ def getJson(json_data):
...
+    def getJsonValue(client, key):
+        try:
+            json_object = json.loads(client.get(key)[0])
+            return json_object
+        except Exception as ex:
+            raise Exception("Getvalue:", key, ex)
+
+    # etcd utility for checking if a key exist
+    @staticmethod
+    def checkIfKeyExist(client, key):
```
Comment (redhood-97) [inline api/util/utilityHelpers.py:51]: Make the same prefix key change here as well.

File: api/routers/odin.py (hunk 1 above)
Comment (redhood-97) [reply api/routers/odin.py:92]: Correction: keep it like this "service:acl_token-{actual-acl-token}"

File: api/routers/odin.py [hunk 3]
```diff
@@ -15,6 +17,8 @@ async def deploy_service(deploy_request: DeployRequest):
         Helm.odinHelmSetup()
         output = Utils.getJson(Helm.deployService(deploy_request.service_name,
                                                   deploy_request.chart_name, deploy_request.values))
+        service_list = Utils.getJson(Helm.listAllServices())
+        etcd.put('{user_id}:service_list'.format(user_id=deploy_request.userId), json.dumps(service_list))
```
Comment (redhood-97) [inline api/routers/odin.py:21]: Make the key change here as mentioned in the subsequent comments.

**Review by redhood-97 — CHANGES_REQUESTED** [review]
Key-change

File: api/routers/odin.py (hunk 1 above)
Comment (TheAlgo) [reply api/routers/odin.py:92]: I don't think we need to be that verbose instead we can keep it as `{acl_token}:service_list`

**Review by redhood-97 — APPROVED** [review]
Looks fine to me !

**Comment by anuragsarkar97** [comment]
there should be no context of acl in odin. 
1. mapping should be on cluster basis. becuase we are currently only looking at one cluster. please store it as is. do not take into consideration acl 
2. if you are going to take in acl plase pass it as a header and not a request param.

**Pull request merged**